swarm/metrics.py

```diff
@@ -0,0 +1,165 @@
+__author__ = "Aidan Morrison <aidandmorrison@gmail.com>"
+
+import numpy as np
+
+
+def summarise_across(data:np.ndarray, func_string: str = 'mean', across: str = "bees") -> np.ndarray:
+    """
+    This function aggregates along the first dimension of a multidimensional array.
+    In this library, it has two main use cases:
+
+    1: This function returns a summary statistic regarding predictions at a particular x point,
+    for a particular epoch.  Hence, it summarises the predictions made across different bees
+    in a swarm.
+
+    2: This function returns a summary statistic regarding the losses of different bees at each epoch.
+
+    3. This function returns a summary statistic regarding of another summary statistic recorded at each
+    point in the domain. In this case it should be used in with argument `across` set to `x`.
+
+    Note: Could use a refactor to generalise so that any one-d function can be passed in
+    as an argument. (didn't do this initially because the range concept was important, and
+    I wanted this to be accessible along the same lines as mean, max, min etc.)
+
+    Args:
+        data: In use case 1 this should be a numpy array of predictions, as yeilded by core.swarm_train.
+            It will have three dimensions, of structure [bee, epoch/batch, x_step].
+            In use case 2 this should be a numpy array of losses, which can also be yeilded by core.swarm_train
+            It will have two dimensions, of structure [bee, epoch].
+            In use case 3 this should be a numpy array of summary statistics across bees at all points in domain x.
+            It will have two dimensions, of structure [epoch/batch, x_step]
+        func_string: A string indicating the type of summary statistic desired.
+            Currently allowable includes 'min', 'max', 'mean', 'median', 'std', and 'range'
+        across: Takes ony 'Indicate whether you'd like to summarise across the bees (as in use case 1 and 2),
+            or summarise across the a value for each point in domain x.
+
+    Returns: a numpy array of two dimensions, of structure [epoch/batch, x_step] that gives the
```

Comment:
shape returned would depend on the `across` result

Reply:
Ah yes.